src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -1372,14 +1376,19 @@ internal bool FilterAndAppendDiagnostics(DiagnosticBag accumulator, IEnumerable<
                     continue;
                 }
 
-                var filtered = Options.FilterDiagnostic(d);
-                if (filtered == null ||
-                    (!reportSuppressedDiagnostics && filtered.IsSuppressed))
+                Diagnostic filtered = d;
+                if (filterDiagnostics)
                 {
-                    continue;
+                    filtered = Options.FilterDiagnostic(d);
```

Comment:
>filtered = Options.FilterDiagnostic(d); [](start = 20, length = 39)

Is there a good reason to make this call conditional? It was unconditional before.

Reply:
Yes, this is required to ensure that the filtering of the diagnostics (i.e. applying pragma, warnaserror, etc.) is deferred when we are executing a build which involves DiagnosticSuppressors. I hope the comment [here](https://github.com/dotnet/roslyn/pull/36067/files/08538a823676aeff914b1f61e978ebf6094f037a#diff-a969f68cd2bc892509eb83bec00d060eR859) would clarify why we need conditional filtering during compilation phases.